portal-frontend/src/app/features/applications/alcs-edit-submission/alcs-edit-submission.component.ts

```diff
@@ -143,7 +146,16 @@ export class AlcsEditSubmissionComponent implements OnInit, OnDestroy, AfterView
 
   // this gets fired whenever applicant navigates away from edit page
   async canDeactivate(): Promise<Observable<boolean>> {
-    await this.saveApplication(this.customStepper.selectedIndex);
+    if (!this.isDeactivating) {
```

Comment:
nitpick: in theory, it should be possible to create a helper function or base component that will have canDiactivate implemented with the try-catch logic and accept save method from the specific component.